seqr_scripts/seqr_sample_qc.py

```diff
@@ -0,0 +1,530 @@
+    }
+    if data_type == "WES":
+        flags.update(
+            {
+                "coverage": mt.HS_PCT_TARGET_BASES_20X
+                < metric_thresholds["wes_cov_thres"]
+            }
+        )
+    else:
+        flags.update(
+            {"coverage": mt.WGS_MEAN_COVERAGE < metric_thresholds["wgs_cov_thres"]}
+        )
+
+    return add_filters_expr(flags)
+
+
+def get_all_sample_metadata(
+    mt: hl.MatrixTable, build: int, data_type: str, data_source: str, version: int
+) -> hl.Table:
+    """
+    Annotate MatrixTable with all current metadata: sample sequencing metrics, sample ID mapping,
+    and callrate for bi-allelic, high-callrate common SNPs.
+    :param MatrixTable mt: VCF converted to a MatrixTable
+    :param int build: build for write, 37 or 38
+    :param str data_type: WGS or WES for write path and flagging metrics
+    :param str data_source: internal or external for write path
+    :param int version: Int for write path
+    :return: Table with seq metrics and mapping
+    :rtype: Table
+    """
+    logger.info("Importing and annotating with sequencing metrics...")
+    meta_ht = hl.import_table(
+        seq_metrics_path(build, data_type, data_source, version), impute=True
+    ).key_by("SAMPLE")
+
+    logger.info("Importing and annotating seqr ID names...")
+    remap_ht = hl.import_table(
+        remap_path(build, data_type, data_source, version), impute=True
+    ).key_by("s")
+    meta_ht = meta_ht.annotate(**remap_ht[meta_ht.key])
+    meta_ht = meta_ht.annotate(
+        seqr_id=hl.if_else(
+            hl.is_missing(meta_ht.seqr_id), meta_ht.SAMPLE, meta_ht.seqr_id
+        )
+    )
+
+    logger.info(
+        "Filtering to bi-allelic, high-callrate, common SNPs to calculate callrate..."
+    )
+    mt = get_qc_mt(
+        mt,
+        apply_hard_filters=False,
+        filter_decoy=False,
+        filter_segdup=False,
+        min_inbreeding_coeff_threshold=None,
+        min_hardy_weinberg_threshold=None,
+        ld_r2=None,
+    )
+    meta_ht = meta_ht.annotate(filtered_callrate=mt.cols()[meta_ht.key].sample_callrate)
+    return meta_ht
+
+
+def run_platform_imputation(
+    mt: hl.MatrixTable,
+    plat_min_cluster_size: int,
+    plat_min_sample_size: int,
+    plat_assignment_pcs: int,
+) -> hl.Table:
+    """
+    Run PCA using sample callrate across gnomAD's evaluation interval and create Hail Table
+    with platform PCs and assigned platform.
+    :param MatrixTable mt: QC MatrixTable
+    :param plat_min_cluster_size: min cluster size for HBDscan clustering
+    :param plat_min_sample_size: min sample size for HBDscan clustering
+    :param plat_assignment_pcs: Number PCs used for HBDscan clustering
+    :return: Table with platform PCs and assigned platform
+    :rtype: Table
+    """
+    intervals = hl.import_locus_intervals(evaluation_intervals_path)
+    callrate_mt = compute_callrate_mt(mt, intervals)
+    eigenvalues, scores_ht, _ = run_platform_pca(callrate_mt)
```

Comment:
`run_platform_pca` has a `binarization_threshold` that sets the callrate to a 0/1 threshold and is set to 0.25 by default. we set it to None in the UKBB code for consistency (we wanted to run platform inference consistently across all the tranches); not sure if this is relevant here?

Reply:
Thanks. I will need to think about the threshold use here considering the large number of technologies/platforms in the callset.